packages/tizen_app_manager/lib/src/app_manager.dart

```diff
@@ -0,0 +1,261 @@
+  appManagerErrorAppNoRunning: 'appManagerErrorAppNoRunning',
+  appManagerErrorRequestFailed: 'appManagerErrorRequestFailed',
+};
+
+String _getErrorCode(int returnCode) => _errorCodes.containsKey(returnCode)
+    ? _errorCodes[returnCode]!
+    : '$returnCode';
+
+AppManager? _appManagerInstance;
+AppManager get appManager => _appManagerInstance ??= AppManager();
+
+class AppManager {
+  AppManager() {
+    final libAppMananger =
+        DynamicLibrary.open('libcapi-appfw-app-manager.so.0');
+    getAppContext = libAppMananger
+        .lookup<NativeFunction<_AppManagerGetAppContext>>(
+            'app_manager_get_app_context')
+        .asFunction();
+    appIsRunning = libAppMananger
+        .lookup<NativeFunction<_AppManagerIsRunning>>('app_manager_is_running')
+        .asFunction();
+    getPackageId = libAppMananger
+        .lookup<NativeFunction<_AppContextGetPackageId>>(
+            'app_context_get_package_id')
+        .asFunction();
+    getProcessId = libAppMananger
+        .lookup<NativeFunction<_AppContextGetPid>>('app_context_get_pid')
+        .asFunction();
+    getAppState = libAppMananger
+        .lookup<NativeFunction<_AppContextGetAppState>>(
+            'app_context_get_app_state')
+        .asFunction();
+    terminateApp = libAppMananger
+        .lookup<NativeFunction<_AppManagerTerminate>>(
+            'app_manager_terminate_app')
+        .asFunction();
+    resumeApp = libAppMananger
+        .lookup<NativeFunction<_AppManagerResumeApp>>('app_manager_resume_app')
+        .asFunction();
+    destroyContext = libAppMananger
+        .lookup<NativeFunction<_AppContextDestroy>>('app_context_destroy')
+        .asFunction();
+  }
+
+  late int Function(Pointer<Utf8>, Pointer<Pointer<_ContextHandle>>)
+      getAppContext;
+  late int Function(Pointer<Utf8>, Pointer<Int8>) appIsRunning;
+  late int Function(Pointer<_ContextHandle>, Pointer<Pointer<Utf8>>)
+      getPackageId;
+  late int Function(Pointer<_ContextHandle>, Pointer<Int32>) getProcessId;
+  late int Function(Pointer<_ContextHandle>, Pointer<Int32>) getAppState;
+  late int Function(Pointer<_ContextHandle>) terminateApp;
+  late int Function(Pointer<_ContextHandle>) resumeApp;
+  late int Function(Pointer<_ContextHandle>) destroyContext;
+}
+
+class AppContext {
+  Pointer<_ContextHandle> _handle = nullptr;
+  int _handleAddress = 0;
+  late String _appId;
+
+  AppContext(String appId, int handleAddress) {
+    if (appId.isEmpty) {
+      throw PlatformException(
+        code: _getErrorCode(tizenErrorInvalidParameter),
+        message: 'appId is empty',
+      );
+    }
+
+    _appId = appId;
+    if (handleAddress == 0) {
+      final Pointer<Pointer<_ContextHandle>> pHandle = malloc();
+      final appIdP = appId.toNativeUtf8();
+      try {
+        final ret = appManager.getAppContext(appIdP, pHandle);
+        if (ret != 0) {
+          throw PlatformException(
+            message: 'Failed to excute app_manager_get_app_context $appId',
```

Comment:
`getErrorMessage(ret)`  is similiar to `_getErrorCode(int ret)` in this code.
common error message is only return simple information about the error. 
so, I want to keep current messages for errors.

Reply:
Okay, but you forgot to fix the typos :)

FYI) url_launcher_tizen used to use the same method as yours to convert platform error codes (https://github.com/flutter-tizen/plugins/commit/349644abbe0653b605bab34c53ceb2379eb2d079#diff-2c65af44c2da2da6e06c7daff9319f63682b8949aa9016ad9c556d314f6d12f5) but I decided to simplify them because they looked messy and it could already be known from the stack trace which function failed without printing the function name in the error message.

Reply:
I will follow your suggestion because it already changed in url_launcher.